### DISCUSSION TITLE
Gradually implement a rule

### DISCUSSION THREAD

**Discussion by jamesmstone** [eslint/eslint#14092] [Q&A]
Hi,

Is it possible to gradually more a rule from "warn" to "error"

For example, when working with a large code base it would be nice to add a rule and only have it apply for files that have been modified after a certain date (or certain commit). So all old files won't cause an error, but rules that would error in new files (or old files that have been changed since this date) do break the build.

**Comment by g-plane** [comment]
While ESLint won't support this complex feature, you can write a script to achieve this by calling ESLint API, so you can filter warnings and errors according to a certain date or commit.

**Reply by jamesmstone** [reply]
Thanks @g-plane, I believe you are reffering to this https://eslint.org/docs/developer-guide/nodejs-api

Do you have any examples of this specific case using git revisions ?

**Reply by g-plane** [reply]
The simple way to achieve this is spawn the `git` command by using `cross-spawn` (or other similar modules) then capture the stdout, so you can gather information from stdout of the `git` command.

**Reply by g-plane** [reply]
Or, you may use [simple-git](https://github.com/steveukx/git-js) module, but I haven't used it so I don't know more about it.

**Reply by jamesmstone** [reply]
Thanks! I will give that a go
